src/index.ts

```diff
@@ -38,14 +65,14 @@ export function parseXY(text, options = {}) {
 
   let lines = text.split(/[\r\n]+/);
 
-  let matrix = [];
-  let info = [];
+  let matrix: number[][] = [];
+  let info: { position: number; value: string }[] = [];
   let position = 0;
-  for (let l = 0; l < lines.length; l++) {
-    let line = lines[l].trim();
+  lines.forEach((line) => {
```

Comment:
That's fine, just want to make you aware here that you could use `for(let line of lines)` as well, which I personally find more readable